### ISSUE TITLE
Promise not returned from lerpLookAt if camera already at final position

### ISSUE THREAD

**Issue by cyrusadkisson** [yomotsu/camera-controls#214]
## Describe the bug

Using the following function call:

```
cameraControls.lerpLookAt(
    xpoint,
    ypoint - 18,
    20 + (elevation * EXTRUSION_FACTOR),
    xpoint,
    ypoint,
    (elevation * EXTRUSION_FACTOR),
    xpoint,
    ypoint - 180,
    200 + (elevation * EXTRUSION_FACTOR),
    xpoint,
    ypoint,
    (elevation * EXTRUSION_FACTOR),
    1,
    true
).then(function() {
    console.log("done");
});
```
"done" appears in the console if you execute this and the lerpLookAt animation occurs and finishes from some arbitrary starting point.

But i you leave the camera at its finishing position and then do this again, the "done" does not appear immediately.

This "done" will only finally show up in the log once you do any other camera animation.

## To Reproduce

Use the code snippet above on a scene. Bind it to a keypress or something. Move the camera manually, use the keypress and watch for "done" which should appear when the transition is finished.

Without touching the camera again, hit the keypress a second time. You'll see that the transition event starts but "done" never shows up...

Until you execute some other animation/transition later.

### Code

```js
see above
```

### Live example

https://www.etheria.world/explore.html

Click a tile, then mouse over the tile and press "c". Look for "done2"
Without moving camera, press "c" again. You won't see "done2" again until another transition occurs and finishes.

**Comment by cyrusadkisson** [comment]
More info: Seems correlated with "rest" event. "done" only appears when "rest" does if you have a rest event listener set. (The live example now does.)

**Comment by yomotsu** [comment]
Thanks for your report.
Your indication is right.

These kinds of methods should check if the final point is already set, resolve immediately just as the same as non-transition mode.

**Comment by cyrusadkisson** [comment]
Ok. Will I be notified here when a fix is merged?

**Comment by yomotsu** [comment]
I'm working on it and complete hopefully this weekend.

**Comment by yomotsu** [comment]
I think this has been solved by #217
